.vsts-pipelines/templates/steps/script.yml

```diff
@@ -0,0 +1,70 @@
+# File: templates/steps/script.yml
+
+parameters:
+  name: 'DO NOT LEAVE DEFAULT'
+  agentOs: 'Windows'
+  beforeBuild: []
+  buildScript: ''
+  buildArgs: ''
+  afterBuild: []
+  hasTests: false
+  autoArtifact: true
+  envVars:
+    BuildNumber: $(Build.BuildNumber)
+    CI: true
+    DOTNET_CLI_TELEMETRY_OPTOUT: 1
+    DOTNET_HOME: '$(Build.SourcesDirectory)/.dotnet'
+    DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+
+steps:
+  - ${{ parameters.beforeBuild }}
+  - ${{ if endsWith(parameters.buildScript, '.ps1') }}:
+    - task: Powershell@1
+      displayName: ${{ parameters.buildScript }}
+      env:
+        ${{ parameters.envVars }}
+      inputs:
+        name:
+        scriptType: filePath
+        scriptName: ${{ parameters.buildScript }}
+        arguments: ${{ parameters.buildArgs }}
+  - ${{ if endsWith(parameters.buildScript, '.sh') }}:
+    - task: Bash@3
+      displayName: ${{ parameters.buildScript }}
+      env:
+        ${{ parameters.envVars }}
+      inputs:
+        scriptType: filePath
+        filePath: ${{ parameters.buildScript }}
+        arguments: ${{ parameters.buildArgs }}
+  - ${{ if endsWith(parameters.buildScript, '.cmd') }}:
+    - task: BatchScript@1
+      displayName: ${{ parameters.buildScript }}
+      env:
+        ${{ parameters.envVars }}
+      inputs:
+        filename: ${{ parameters.buildScript }}
+        arguments: ${{ parameters.buildArgs }}
+  - ${{ if eq(parameters.hasTests, true) }}:
+    - task: PublishTestResults@2
+      displayName: Publish test results
+      condition: always()
+      inputs:
+        testRunner: vstest
+        testResultsFiles: 'modules/*/artifacts/logs/*.trx'
+  - ${{ parameters.afterBuild }}
+  - ${{ if eq(parameters.autoArtifact, true) }}:
+    - task: PublishBuildArtifacts@1
```

Comment:
Can we make artifacts conditional on whether or not this is a public or private build? Generally speaking, we do not want to publish artifacts from public builds, only internal-official ones.